### ISSUE TITLE
Incorrect results in disk_encryption table for AES-XTS encryption

### ISSUE THREAD

**Issue by kliseb** [osquery/osquery#4658]
### What version of `osquery` are you using? What operating system. Be specific.
osquery_version = 3.2.6
os_version = 10.13.5
os_build = 17F77
os_platform = darwin
### What have you already tried?
I am attempting to get the encryption status of an external drive with
```
osquery> select * from disk_encryption;
```
### What did you expect to see?
```
+--------------+--------------------------------------+-----------+---------+-----+-----------+
| name         | uuid                                 | encrypted | type    | uid | user_uuid |
+--------------+--------------------------------------+-----------+---------+-----+-----------+
|       .      |                  .                   |     .     |    .    |  .  |     .     |
|       .      |                  .                   |     .     |    .    |  .  |     .     |
|       .      |                  .                   |     .     |    .    |  .  |     .     |
| /dev/disk3   | BE1ACE01-5E00-431F-A49B-1E77D699ECFF | 1         | AES-XTS |     |           |
+--------------+--------------------------------------+-----------+---------+-----+-----------+
```
### What did you see instead?
```
E0628 11:05:35.418220 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.419442 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.420858 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.422544 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.531699 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.532852 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.533974 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.535043 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
E0628 11:05:35.535866 2507449216 disk_encryption.mm:259] Error calling isEncryptedVolume:encrypted:
+--------------+--------------------------------------+-----------+-----------------+-----+--------------------------------------+
| name         | uuid                                 | encrypted | type            | uid | user_uuid                            |
+--------------+--------------------------------------+-----------+-----------------+-----+--------------------------------------+
| /dev/disk0   |                                      |           |                 |     |                                      |
| /dev/disk0s1 | 0F533033-75F3-4AB4-884B-882678B41BAE |           |                 |     |                                      |
| /dev/disk0s2 | 71404D30-D9E1-4014-A51E-35C4EBCECBFE |           |                 |     |                                      |
| /dev/disk1   | EA5A0C41-0412-4730-92F2-CCAAADFE4799 |           |                 |     |                                      |
| /dev/disk1s1 | AFB525D1-0EF0-4211-BDF0-F74D9ABE0451 | 1         | APFS Encryption | 501 | 43D11658-BE39-4197-9A95-499D72FE1FB4 |
| /dev/disk1s2 | 7A20B06D-C91C-4756-8A2F-33A6E66B83F3 | 0         |                 |     |                                      |
| /dev/disk1s3 | 2AF78E0E-5BC4-42C1-915E-80BFD89EE1FC | 0         |                 |     |                                      |
| /dev/disk1s4 | 13E0A23A-1FF2-4593-9795-F3593E87B62C | 0         |                 |     |                                      |
| /dev/disk2   |                                      |           |                 |     |                                      |
| /dev/disk2s1 | 83FA93EF-FFEE-4EB4-8914-0748803EF796 |           |                 |     |                                      |
| /dev/disk2s2 | BFE77258-4A09-4299-B8BE-95B1B224D9CD |           |                 |     |                                      |
| /dev/disk2s3 | D0850E11-C5EF-4C18-8F2F-48286B8F8715 |           |                 |     |                                      |
| /dev/disk3   | BE1ACE01-5E00-431F-A49B-1E77D699ECFF |           |                 |     |                                      |
+--------------+--------------------------------------+-----------+-----------------+-----+--------------------------------------+
```
The table showed the correct result for the external drive with osquery version 2.8.0:
```
brandonklise@brandons-macbook-pro: /tmp/2.8
$ ./usr/local/bin/osqueryi --version
osqueryi version 2.8.0
brandonklise@brandons-macbook-pro: /tmp/2.8
$ ./usr/local/bin/osqueryi
Using a virtual database. Need help, type '.help'
osquery> select * from disk_encryption;
+--------------+--------------------------------------+-----------+---------+-----+-----------+
| name         | uuid                                 | encrypted | type    | uid | user_uuid |
+--------------+--------------------------------------+-----------+---------+-----+-----------+
| /dev/disk0   |                                      | 0         |         |     |           |
| /dev/disk0s1 | 0F533033-75F3-4AB4-884B-882678B41BAE | 0         |         |     |           |
| /dev/disk0s2 | 71404D30-D9E1-4014-A51E-35C4EBCECBFE | 0         |         |     |           |
| /dev/disk1   | EA5A0C41-0412-4730-92F2-CCAAADFE4799 | 0         |         |     |           |
| /dev/disk1s1 | AFB525D1-0EF0-4211-BDF0-F74D9ABE0451 | 0         |         |     |           |
| /dev/disk1s2 | 7A20B06D-C91C-4756-8A2F-33A6E66B83F3 | 0         |         |     |           |
| /dev/disk1s3 | 2AF78E0E-5BC4-42C1-915E-80BFD89EE1FC | 0         |         |     |           |
| /dev/disk1s4 | 13E0A23A-1FF2-4593-9795-F3593E87B62C | 0         |         |     |           |
| /dev/disk2   |                                      | 0         |         |     |           |
| /dev/disk2s1 | 83FA93EF-FFEE-4EB4-8914-0748803EF796 | 0         |         |     |           |
| /dev/disk2s2 | BFE77258-4A09-4299-B8BE-95B1B224D9CD | 0         |         |     |           |
| /dev/disk2s3 | D0850E11-C5EF-4C18-8F2F-48286B8F8715 | 0         |         |     |           |
| /dev/disk3   | BE1ACE01-5E00-431F-A49B-1E77D699ECFF | 1         | AES-XTS |     |           |
+--------------+--------------------------------------+-----------+---------+-----+-----------+
```
However, this was broken in #3748:
```
brandonklise@brandons-macbook-pro: /tmp/2.9
$ ./usr/local/bin/osqueryi --version
osqueryi version 2.9.0
brandonklise@brandons-macbook-pro: /tmp/2.9
$ ./usr/local/bin/osqueryi
Using a virtual database. Need help, type '.help'
osquery> select * from disk_encryption;
+--------------+--------------------------------------+-----------+-----------------+-----+--------------------------------------+
| name         | uuid                                 | encrypted | type            | uid | user_uuid                            |
+--------------+--------------------------------------+-----------+-----------------+-----+--------------------------------------+
| /dev/disk0   |                                      | 0         |                 |     |                                      |
| /dev/disk0s1 | 0F533033-75F3-4AB4-884B-882678B41BAE | 0         |                 |     |                                      |
| /dev/disk0s2 | 71404D30-D9E1-4014-A51E-35C4EBCECBFE | 0         |                 |     |                                      |
| /dev/disk1   | EA5A0C41-0412-4730-92F2-CCAAADFE4799 | 0         |                 |     |                                      |
| /dev/disk1s1 | AFB525D1-0EF0-4211-BDF0-F74D9ABE0451 | 1         | APFS Encryption | 501 | 43D11658-BE39-4197-9A95-499D72FE1FB4 |
| /dev/disk1s2 | 7A20B06D-C91C-4756-8A2F-33A6E66B83F3 | 0         |                 |     |                                      |
| /dev/disk1s3 | 2AF78E0E-5BC4-42C1-915E-80BFD89EE1FC | 0         |                 |     |                                      |
| /dev/disk1s4 | 13E0A23A-1FF2-4593-9795-F3593E87B62C | 0         |                 |     |                                      |
| /dev/disk2   |                                      | 0         |                 |     |                                      |
| /dev/disk2s1 | 83FA93EF-FFEE-4EB4-8914-0748803EF796 | 0         |                 |     |                                      |
| /dev/disk2s2 | BFE77258-4A09-4299-B8BE-95B1B224D9CD | 0         |                 |     |                                      |
| /dev/disk2s3 | D0850E11-C5EF-4C18-8F2F-48286B8F8715 | 0         |                 |     |                                      |
| /dev/disk3   | BE1ACE01-5E00-431F-A49B-1E77D699ECFF | 0         |                 |     |                                      |
+--------------+--------------------------------------+-----------+-----------------+-----+--------------------------------------+
```

**Comment by terracatta** [comment]
cc/ @mkareta did you ever PR https://github.com/mkareta/osquery/commit/73a38c0abce95bfa878890e4d6ac0d76b009fd82 ?

**Comment by terracatta** [comment]
Related or dupe of https://github.com/facebook/osquery/issues/4479

**Comment by mkareta** [comment]
Unfortunately, It's unrelated to #4479 
@kliseb I will try to look at it next week, meanwhile can you please do "diskutil info -all" and copy info about that partition, thanks!

**Comment by kliseb** [comment]
I'm on my way to go backpacking and won't have my computer with me this weekend. I'll post the info about the partition when I get back late Sunday or early Monday.

**Comment by kliseb** [comment]
```
   Device Identifier:        disk3
   Device Node:              /dev/disk3
   Whole:                    Yes
   Part of Whole:            disk3
   Device / Media Name:      Elements 25A2

   Volume Name:              Backup
   Mounted:                  Yes
   Mount Point:              /Volumes/Backup

   Content (IOContent):      Apple_HFSX
   File System Personality:  Case-sensitive Journaled HFS+
   Type (Bundle):            hfs
   Name (User Visible):      Mac OS Extended (Case-sensitive, Journaled)
   Journal:                  Journal size 155648 KB at offset 0x3a35000
   Owners:                   Enabled

   OS Can Be Installed:      Yes
   Booter Disk:              disk2s3
   Media Type:               Generic
   Protocol:                 USB
   SMART Status:             Not Supported
   Volume UUID:              385B15EB-052F-38A3-AB89-AD0C3FC2AC4B
   Disk / Partition UUID:    BE1ACE01-5E00-431F-A49B-1E77D699ECFF

   Disk Size:                2.0 TB (1999659597824 Bytes) (exactly 3905585152 512-Byte-Units)
   Device Block Size:        512 Bytes

   Volume Total Space:       2.0 TB (1999659597824 Bytes) (exactly 3905585152 512-Byte-Units)
   Volume Used Space:        30.0 GB (29974388736 Bytes) (exactly 58543728 512-Byte-Units) (1.5%)
   Volume Free Space:        2.0 TB (1969685209088 Bytes) (exactly 3847041424 512-Byte-Units) (98.5%)
   Allocation Block Size:    4096 Bytes

   Read-Only Media:          No
   Read-Only Volume:         No

   Device Location:          External
   Removable Media:          Fixed

   Virtual:                  Yes

   This disk is a Core Storage Logical Volume (LV).  Core Storage Information:
   LV UUID:                  BE1ACE01-5E00-431F-A49B-1E77D699ECFF
   LVF UUID:                 307874E6-2201-45F8-89F9-345BA8DF1B5F
   LVG UUID:                 B6630286-FE97-42EE-B654-DC1699190FCC
   PV UUID (disk):           755B3A36-CD5F-412F-BC52-2892020667E7 (disk2s2)
   Fusion Drive:             No
   Encrypted:                Yes
```

**Comment by directionless** [comment]
This looks like it was fixed in #4691
